### ISSUE TITLE
Question about validation_middleware

### ISSUE THREAD

**Issue by korcky** [maximdanilchenko/aiohttp-apispec#64]
Currently code of this method looks like:
```python
async def validation_middleware(request: web.Request, handler) -> web.Response:
    ...
    for schema in schemas:
        data = await request.app["_apispec_parser"].parse(
            schema["schema"], request, locations=schema["locations"]
        )
        if data:
            try:
                result.update(data)
            except (ValueError, TypeError):
                result = data
                break
    else:
        result.update(request.match_info)
    request[request.app["_apispec_request_data_name"]] = result
    return await handler(request)
```
So, if schema successfully parsed a request arguments (basically transform each argument to their relative type), they will be put in `result`, but eventually (if `data` a dictionary) arguments in `result` will be replaced by their string representation.

The question is `Why?` Why replacing a right type arguments with their string representation?

**Comment by maximdanilchenko** [comment]
@korcky, you talk about usage of `@request_schema` with `location="match_info"`, am I right?

**Comment by korcky** [comment]
@maximdanilchenko, yep, you're correct

**Comment by korcky** [comment]
I'm not quite affected by this issue for now) So, I think it would better to fix in 2.0 version.
